stdlib/src/memory/unsafe_pointer.mojo

```diff
@@ -624,6 +628,7 @@ struct UnsafePointer[
             offset: The offset to store to.
             val: The value to store.
         """
+        constrained[is_mutable, _must_be_mut_err]()
```

Comment:
Can we turn these into conditional conformance?

Reply:
Theoretically the compiler won't even let you mutate the origin. I added them for convenience so that users get a clear error message. Conditional Conformance doesn't give you that, only a cryptic error that some parameters can't be inferred.